src/modules/fw_path_navigation/FixedwingPositionControl.cpp

```diff
@@ -2364,35 +2235,6 @@ FixedwingPositionControl::reset_landing_state()
 	}
 }
 
-bool FixedwingPositionControl::using_npfg_with_wind_estimate() const
-{
-	// high wind mitigation is handled by NPFG if the wind estimate is valid
-	return _param_fw_use_npfg.get() && _wind_valid;
-}
-
-Vector2f
-FixedwingPositionControl::get_nav_speed_2d(const Vector2f &ground_speed)
```

Comment:
why do you delete this method? i dont particularly like the air/ground angle logic, is a bit clunky, but it is the last defense in high wind when we dont have wind estimate. of course, one could also argue that if we have a valid airspeed estimate.. we should probably still have valid wind (or maybe not if GNSS is out?)

Reply:
That,s what i wanted to discuss with you, but i thought this was only done because L1 would get unstable, when we move "backwards" with respect to ground. In this case, it replaces the ground speed with the airspeed, but this gets then fed into the npfg navigate functions as ground speed again?!? Doesn't that screw up the npfg calculations? Also the npfg does internally account for that in the course to heading conversion or not? Also, this function is not always used, but for me only at random places.

Reply:
>  but i thought this was only done because L1 would get unstable, when we move "backwards" with respect to ground

yes, but we're considering a case here where there is no wind estimate, so the ground speed is the only thing getting input to NPFG. So effectively NPFG is no better than L1.. because it doesnt think there is any wind.

> but this gets then fed into the npfg navigate functions as ground speed again?!?

yeah it's a hack. but results in something similar to what NPFG does for backwards flight.

Now that I re-think about it though .. if we have an airspeed measurement, and a ground velocity estimate.. there is ZERO reason we shouldn't have at least a cursory wind estimate.

I'm good with removing the `nav_speed_2d()` method then. But then we need to check what we do with no wind estimate. What I'd propose...
- clarify if there is ever a situation where we have airspeed yaw and GNSS measurements, but no wind estimate @sfuhrer @bresch - if there is .. we can consider how to handle this (e.g. if we combine the measurements ourselves in the fw pos ctrl.. but that seems silly)
- in the case of no valid wind estimate (this ideally is only ever true when we are in AIRSPEEDLESS mode), feed in ground speed directly with zero for the wind vector. But then also adapt the minimum ground speed settings in NPFG (if they are not already set to some non-zero value). Then NPFG will do its best to maintain some non-zero ground speed despite not having the yaw/airspeed vector.
- beyond this, we cannot do any better, but at least we don't need the additional nav_speed_2d method.

Reply:
>clarify if there is ever a situation where we have airspeed yaw and GNSS measurements, but no wind estimate @sfuhrer @bresch - if there is .. we can consider how to handle this (e.g. if we combine the measurements ourselves in the fw pos ctrl.. but that seems silly)

Currently the EKF wind estimate can be switched off, and it's hard to prevent that unless we eg have beta fusion forcefully enabled always, and EKF2_ARSP_THR set below FW_AIRSPD_TRIM or so.
If we want to be independent of the EKF wind estimates then we could add another standalone wind estimator within the FW guidance module, which we then could also independently tune and keep enabled always. 
I would rather rely on the EKF wind estimate, tune it to be more dynamic and at least give a warning if EKF2_ARSP_THR set below FW_AIRSPD_TRIM and EKF2_FUSE_BETA is disabled.

Reply:
I'd prefer not to re-invent the wheel with state estimation inside the controller.. since it should already be feasible in the EKF. Also.. don't like the idea of using the measurements directly (compatibility / consistency for feedback states to the controller).

So for the case that EKF wind estimate is (for whatever reason) turned off - I suppose this would mean assuming zero wind and bumping the min ground speed to non-zero value. Best then we could do is advise.. don't turn off the wind estimate..?

Otherwise we are just needlessly re-implementing a worse wind estimation in the controller if we use the measurements directly.

wdyt @sfuhrer @KonradRudin ?

Reply:
I don't see a reason why a wind estimator in here would have worse performance, but I don't like the complexity, load and weight it adds. That's why my recommendation is the one noted above 
"I would rather rely on the EKF wind estimate, tune it to be more dynamic and at least give a warning if EKF2_ARSP_THR set below FW_AIRSPD_TRIM and EKF2_FUSE_BETA is disabled."
The param check and warning could reside in the FW guidance module. 
EKF2_WIND_NSD then should be increased, and the wind estimate instead be low passed filtered locally where required (eg for the "wind speed high" warning/RTL).

Reply:
> EKF2_WIND_NSD then should be increased, and the wind estimate instead be low passed filtered locally where required (eg for the "wind speed high" warning/RTL).

yes pls.

> give a warning if EKF2_ARSP_THR set below FW_AIRSPD_TRIM and EKF2_FUSE_BETA is disabled

that sounds fine. - but just to be clear, in this case that means the wind estimate is zero? or just possibly not a great estimate?

> I don't see a reason why a wind estimator in here would have worse performance

It doesn't *have to have worse performance. it's just not necessary to implemented it again in the fixed-wing controller. (especially if doing what is there now.. where it literally just takes the filtered airspeed value and yaw measurement to do a cursory geometry calculation .. so no filtering for the wind estimate)

> So for the case that EKF wind estimate is (for whatever reason) turned off

This is the only open question. What we do when there is no wind estimate. Again, my contention would be.. do nothing except perhaps (if enabled) bump the min ground speed. But I'd like wind use in the controller be intentional on config. So explicitly saying "yes I want wind estimates on" "yes i want to use it in the controller" "yes i understand that if i turn it off.. the controller may do weird turn around things in the very low ground speed flight regime if I dont enable some minimum ground speed"

Reply:
> So instead of removing the function above we should make sure the ground speed is always positive?

I rethought about this - would actually not even do that. We just need to make it clear in documentation what the controller will do with what configuration. So if the user purposely does not set a positive minimum ground speed, and purposely turns off the wind estimate, I would say this is on them. 

TL;DR - would remove the method, and if wind is invalid, set to zero.

Reply:
So did we settle on removing the method here, and propose two things in a second PR:
- increase EKF wind process noise
- add check in FW controller to have EKF wind estimation enabled

Reply:
^strong yes from my side.